pkg/controller/certificaterequests/selfsigned/checks.go

```diff
@@ -85,10 +87,18 @@ func certificateRequestsForSecret(log logr.Logger,
 	dbg.Info("checking if self signed certificate requests reference secret")
 	var affected []*cmapi.CertificateRequest
 	for _, request := range requests {
+		if request.Spec.IssuerRef.Group != cmdoc.GroupName {
+			dbg.Info("issuer has different group indicating it isn't built-in, skipping")
+			continue
+		}
```

Comment:
I think this check is sufficient to close the issue. The IssuerRef.Kind is already enforced by the validating webhook, here:

https://github.com/cert-manager/cert-manager/blob/ffcd374e143e1d859b4cd343cfafb87d2aa8ef37/internal/apis/certmanager/validation/certificate.go#L165-L181


I suggest removing the new error type and the checks and tests below and make only this change.

There is a chance that the validation webhook has been disabled, in which case, that validation function could (should) also be called here and an error logged. 
If the group is cert-manager.io and the Kind is not recognised it should result in an error rather than an info log message.

Reply:
Aaaah, didn't realise that was checked by the validating webhook. I'd assumed it would be possible for an external issuer to create an issuer type with the `cert-manager.io` group.

If that's the case then agreed, I don't think the extra error is needed! I'll make that change, thanks :grin: